src/helpers/figures/charts/bubble_chart.ts

```diff
@@ -0,0 +1,329 @@
+      ? adaptStringRange(chartSheetId, definition.yRange, applyChange)
+      : undefined;
+    const adaptedLabelRange = definition.labelRange
+      ? adaptStringRange(chartSheetId, definition.labelRange, applyChange)
+      : undefined;
+    return {
+      ...definition,
+      type: "bubble",
+      xRange:
+        adaptedXRange && adaptedXRange !== CellErrorType.InvalidReference
+          ? adaptedXRange
+          : undefined,
+      sizeRange:
+        adaptedSizeRange && adaptedSizeRange !== CellErrorType.InvalidReference
+          ? adaptedSizeRange
+          : undefined,
+      yRange:
+        adaptedYRange && adaptedYRange !== CellErrorType.InvalidReference
+          ? adaptedYRange
+          : undefined,
+      labelRange:
+        adaptedLabelRange && adaptedLabelRange !== CellErrorType.InvalidReference
+          ? adaptedLabelRange
+          : undefined,
+    };
+  }
+
+  static getDefinitionFromContextCreation(context: ChartCreationContext): BubbleChartDefinition {
+    return {
+      background: context.background,
+      yRange: context.range?.[0]?.dataRange ?? "",
+      legendPosition: context.legendPosition ?? "top",
+      title: context.title || { text: "" },
+      type: "bubble",
+      labelRange: context.auxiliaryRange || undefined,
+      xRange: context.bubbleXRange || undefined,
+      sizeRange: context.bubbleSizeRange || undefined,
+      axesDesign: context.axesDesign,
+      showValues: context.showValues,
+      humanize: context.humanize,
+      bubbleColor: context.bubbleColorMode || "multiple",
+      verticalAxisPosition: "left",
+    };
+  }
+
+  getDefinition(): BubbleChartDefinition {
+    return this.getDefinitionWithSpecificDataSets(
+      this.yRange,
+      this.labelRange,
+      this.xRange,
+      this.sizeRange
+    );
+  }
+
+  private getDefinitionWithSpecificDataSets(
+    yRange: Range | undefined,
+    labelRange: Range | undefined,
+    xRange: Range | undefined,
+    sizeRange: Range | undefined,
+    targetSheetId?: UID
+  ): BubbleChartDefinition {
+    return {
+      type: "bubble",
+      background: this.background,
+      legendPosition: this.legendPosition,
+      labelRange: labelRange
+        ? this.getters.getRangeString(labelRange, targetSheetId || this.sheetId)
+        : undefined,
+      xRange: xRange
+        ? this.getters.getRangeString(xRange, targetSheetId || this.sheetId)
+        : undefined,
+      sizeRange: sizeRange
+        ? this.getters.getRangeString(sizeRange, targetSheetId || this.sheetId)
+        : undefined,
+      yRange: yRange
+        ? this.getters.getRangeString(yRange, targetSheetId || this.sheetId)
+        : undefined,
+      title: this.title,
+      axesDesign: this.axesDesign,
+      showValues: this.showValues,
+      humanize: this.humanize,
+      bubbleColor: this.bubbleColor,
+      verticalAxisPosition: this.verticalAxisPosition,
+    };
+  }
+
+  getContextCreation(): ChartCreationContext {
+    const ranges = this.yRange
```

Comment:
same as transformDefinition, create a helper function to not repeat 4x the same thing